src/driver/sqlite-abstract/AbstractSqliteDriver.ts

```diff
@@ -507,11 +507,11 @@ export abstract class AbstractSqliteDriver implements Driver {
     /**
      * Creates generated map of values generated or returned by database after INSERT query.
      */
-    createGeneratedMap(metadata: EntityMetadata, insertResult: any) {
+    createGeneratedMap(metadata: EntityMetadata, insertResult: any, entityIndex: number, entityNum: number) {
         const generatedMap = metadata.generatedColumns.reduce((map, generatedColumn) => {
             let value: any;
             if (generatedColumn.generationStrategy === "increment" && insertResult) {
-                value = insertResult;
+                value = insertResult - entityNum + entityIndex + 1;
```

Comment:
can we have the same description like in mysql driver, so we can understand what INSERT statement returns and why we need to deduct entityNum and add entityIndex + 1 ?

From what I understood in case of sqlite insertResult is an id number for the **last** inserted row returned by sqlite?

Reply:
@pleerock Yes. [see here](https://github.com/mapbox/node-sqlite3/wiki/API#databaserunsql-param--callback)
I added a note.